src/test/java/com/powsybl/caseserver/CaseControllerTest.java

```diff
@@ -0,0 +1,174 @@
+/**
+ * Copyright (c) 2019, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.caseserver;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.iidm.xml.NetworkXml;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.mock.web.MockMultipartFile;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.web.servlet.config.annotation.EnableWebMvc;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Map;
+
+import static com.powsybl.caseserver.CaseConstants.*;
+import static org.junit.Assert.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+/**
+ * @author Abdelsalem Hedhili <abdelsalem.hedhili at rte-france.com>
+ */
+
+@RunWith(SpringRunner.class)
+@EnableWebMvc
+@WebMvcTest(CaseController.class)
+@ContextConfiguration(classes = CaseController.class)
+public class CaseControllerTest {
+
+    @Autowired
+    private MockMvc mvc;
+
+    private static final String TEST_CASE = "testCase.xiidm";
+
+    private static final String GET_CASE_URL = "/v1/cases/{caseName}";
+
+    @BeforeClass
+    public static void setup() {
+        final Path path = Paths.get(System.getProperty(USERHOME) + CASE_FOLDER);
```

Comment:
Unit test should never rely on "external" file system but should use jimfs in memory nio2 file system instead.

Reply:
You have to find a way to build CaseService  either with local file system in normal operation, ```FileSystems.getDefault()``` or with a JimsFs.new... for unit testing.